File: manim-videos/custom/colors.py
"""
Color palette for VoxFormer STT visualization
Matching the project's Tailwind theme
"""

from manim import *

# Primary colors (matching Tailwind slate theme)
SLATE_50 = "#f8fafc"
SLATE_100 = "#f1f5f9"
SLATE_200 = "#e2e8f0"
SLATE_300 = "#cbd5e1"
SLATE_400 = "#94a3b8"
SLATE_500 = "#64748b"
SLATE_600 = "#475569"
SLATE_700 = "#334155"
SLATE_800 = "#1e293b"
SLATE_900 = "#0f172a"
SLATE_950 = "#020617"

# Accent colors
CYAN_400 = "#22d3ee"
CYAN_500 = "#06b6d4"
CYAN_600 = "#0891b2"

PURPLE_400 = "#c084fc"
PURPLE_500 = "#a855f7"
PURPLE_600 = "#9333ea"

EMERALD_400 = "#34d399"
EMERALD_500 = "#10b981"
EMERALD_600 = "#059669"

AMBER_400 = "#fbbf24"
AMBER_500 = "#f59e0b"
AMBER_600 = "#d97706"

ROSE_400 = "#fb7185"
ROSE_500 = "#f43f5e"
ROSE_600 = "#e11d48"

BLUE_400 = "#60a5fa"
BLUE_500 = "#3b82f6"
BLUE_600 = "#2563eb"

# Semantic colors
PRIMARY = CYAN_500
SECONDARY = PURPLE_500
ACCENT = EMERALD_500
BACKGROUND = SLATE_900
SURFACE = SLATE_800
TEXT_PRIMARY = SLATE_50
TEXT_SECONDARY = SLATE_400
TEXT_MUTED = SLATE_500

# Audio pipeline colors
WAVEFORM_COLOR = CYAN_400
SPECTROGRAM_LOW = SLATE_900
SPECTROGRAM_HIGH = CYAN_400
MEL_FILTER_COLOR = EMERALD_400
STFT_WINDOW_COLOR = AMBER_400

# Transformer colors
QUERY_COLOR = CYAN_400
KEY_COLOR = PURPLE_400
VALUE_COLOR = EMERALD_400
ATTENTION_COLOR = AMBER_400
FFN_COLOR = ROSE_400
CONV_COLOR = BLUE_400

# Training colors
LOSS_COLOR = ROSE_400
GRADIENT_COLOR = PURPLE_400
WEIGHT_COLOR = EMERALD_400

# Gradient definitions for heatmaps
def get_spectrogram_gradient():
    """Returns color gradient for spectrogram visualization"""
    return [SLATE_900, SLATE_700, CYAN_600, CYAN_400, SLATE_50]

def get_attention_gradient():
    """Returns color gradient for attention weights"""
    return [SLATE_900, PURPLE_600, PURPLE_400, AMBER_400, SLATE_50]

def get_loss_gradient():
    """Returns color gradient for loss visualization"""
    return [EMERALD_500, AMBER_500, ROSE_500]

# Helper function to interpolate colors
def interpolate_color(color1, color2, t):
    """Interpolate between two hex colors"""
    c1 = color_to_rgb(color1)
    c2 = color_to_rgb(color2)
    result = [c1[i] + t * (c2[i] - c1[i]) for i in range(3)]
    return rgb_to_color(result)

File: remotion/src/utils/colors.ts
// Color palette for the Architecture Video
// Matches the existing slide presentation themes

export const COLORS = {
  // Background colors
  slate950: '#020617',
  slate900: '#0f172a',
  slate800: '#1e293b',
  slate700: '#334155',
  slate600: '#475569',

  // VoxFormer theme (Cyan/Purple)
  cyan400: '#22d3ee',
  cyan500: '#06b6d4',
  cyan600: '#0891b2',
  purple400: '#c084fc',
  purple500: '#a855f7',
  purple600: '#9333ea',

  // RAG theme (Emerald/Cyan)
  emerald400: '#34d399',
  emerald500: '#10b981',
  emerald600: '#059669',

  // Avatar theme (Rose/Pink)
  rose400: '#fb7185',
  rose500: '#f43f5e',
  rose600: '#e11d48',
  pink400: '#f472b6',
  pink500: '#ec4899',
  fuchsia500: '#d946ef',

  // MCP theme (Orange/Amber)
  orange400: '#fb923c',
  orange500: '#f97316',
  orange600: '#ea580c',
  amber400: '#fbbf24',
  amber500: '#f59e0b',
  yellow500: '#eab308',

  // Accent colors
  white: '#ffffff',
  black: '#000000',
  green500: '#22c55e',  // Success/Checkmarks
  red500: '#ef4444',    // Errors
  blue500: '#3b82f6',   // Links/Info

  // Transparent variants
  whiteAlpha: (alpha: number) => `rgba(255, 255, 255, ${alpha})`,
  blackAlpha: (alpha: number) => `rgba(0, 0, 0, ${alpha})`,
  cyanAlpha: (alpha: number) => `rgba(6, 182, 212, ${alpha})`,
  emeraldAlpha: (alpha: number) => `rgba(16, 185, 129, ${alpha})`,
  roseAlpha: (alpha: number) => `rgba(244, 63, 94, ${alpha})`,
  orangeAlpha: (alpha: number) => `rgba(249, 115, 22, ${alpha})`,
};

// Component-specific color schemes
export const THEMES = {
  voxformer: {
    primary: COLORS.cyan500,
    secondary: COLORS.purple500,
    accent: COLORS.cyan400,
    glow: COLORS.cyanAlpha(0.3),
  },
  rag: {
    primary: COLORS.emerald500,
    secondary: COLORS.cyan500,
    accent: COLORS.emerald400,
    glow: COLORS.emeraldAlpha(0.3),
  },
  avatar: {
    primary: COLORS.rose500,
    secondary: COLORS.pink500,
    accent: COLORS.rose400,
    glow: COLORS.roseAlpha(0.3),
  },
  mcp: {
    primary: COLORS.orange500,
    secondary: COLORS.amber500,
    accent: COLORS.orange400,
    glow: COLORS.orangeAlpha(0.3),
  },
};

// Gradient definitions
export const GRADIENTS = {
  background: `linear-gradient(135deg, ${COLORS.slate950} 0%, ${COLORS.slate900} 50%, ${COLORS.slate950} 100%)`,
  voxformer: `linear-gradient(135deg, ${COLORS.cyan500} 0%, ${COLORS.purple500} 100%)`,
  rag: `linear-gradient(135deg, ${COLORS.emerald500} 0%, ${COLORS.cyan500} 100%)`,
  avatar: `linear-gradient(135deg, ${COLORS.rose500} 0%, ${COLORS.pink500} 100%)`,
  mcp: `linear-gradient(135deg, ${COLORS.orange500} 0%, ${COLORS.amber500} 100%)`,
  rainbow: `linear-gradient(135deg, ${COLORS.cyan500} 0%, ${COLORS.purple500} 25%, ${COLORS.rose500} 50%, ${COLORS.orange500} 75%, ${COLORS.emerald500} 100%)`,
};
